mimic/model/identity.py

```diff
@@ -20,3 +26,205 @@ def from_store(cls, session_store, tenant_id):
 
     identity_admin = attr(default=False,
                           validator=validators.instance_of(bool))
+
+
+class ICredentials(Interface):
+    """
+    An :obj:`ICredentials` provides identity authentication credentials.
+    """
+    def get_session(session_store):  # pragma:nocover
+        """
+        Get a session corresponding to the user and tenant from the given
+        session store.
+
+        :param session_store: a :class:`mimic.session.SessionStore`
+        """
+
+
+@implementer(ICredentials)
+@attributes
+class PasswordCredentials(object):
+    """
+    An object representing username/password + optional tenant credentials
+    """
+    username = attr(validator=validators.instance_of(string_types))
+    password = attr(validator=validators.instance_of(string_types))
+    tenant_id = attr(default=None)
+
+    def get_session(self, session_store):
+        """
+        Get a session corresponding to the user and tenant from the given
+        session store.
+
+        :param session_store: a :class:`mimic.session.SessionStore`
+        """
+        return session_store.session_for_username_password(
+            self.username, self.password, self.tenant_id)
+
+    @classmethod
+    def from_json(cls, json_blob):
+        """
+        Given an authentication JSON blob, which should look like:
```

Comment:
On the "ReST versus Markdown" front, I think you need a second `:` at the end of the line here to make the next section a code block.